### PR TITLE
Reset result in bulk upgrade loops

### PR DESCRIPTION
**Pull request by pwtyler**
When bulk updating plugins or themes, the result of a previous update remained through the loop, and if a subsequent update failed (which should have then been `null` result), the failed update would end up with the result of the previous operation.

This PR unsets the result at the top of the bulk upgrade loop if it is not already null to prevent the results of the first plugin or theme's success from cascading through a subsequent failure and resulting in a false positive on an update result.

Trac ticket: https://core.trac.wordpress.org/ticket/53002

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-admin/includes/class-plugin-upgrader.php [hunk 1]
```diff
@@ -317,6 +317,8 @@ public function bulk_upgrade( $plugins, $args = array() ) {
 		$this->update_count   = count( $plugins );
 		$this->update_current = 0;
 		foreach ( $plugins as $plugin ) {
+			unset( $this->result );
```
Comment (afragen) [inline src/wp-admin/includes/class-plugin-upgrader.php:320]: What does this do to line 350, now 352 below?

Comment (pwtyler) [reply src/wp-admin/includes/class-plugin-upgrader.php:320]: `$this->result` is set by `this->run()` at line 339-350 (or is at least populated after? I could be missing some specific nuance here), but with the current behavior (before without these changes), on subsequent iterations though the loop, `$this->result` holds the output of the previous update, and if the secondary update fails, $this->result is still the output of update 1, so then `$results[ <plugin2> ]` ends up with faulty information.

To see this behavior, you can add print_r($results) after the whole loop and try the following:
```
$ wp plugin install akismet --version=4.0 --force
$ wp plugin install wordpress-importer --version=0.5 --force
$ chmod -w wp-content/plugins/wordpress-importer
$ wp plugin update --all
# > See "Success: Updated 2 of 2 plugins."
```

Comment (pwtyler) [reply src/wp-admin/includes/class-plugin-upgrader.php:320]: There could be a better place/way to be setting or resetting `$this->result` that I should be using though? 🤷🏻

**Comment by SergeyBiryukov** [comment]
Thanks for the PR! An alternative approach was merged in https://core.trac.wordpress.org/changeset/51528.

**Pull request closed without merging**